### PR TITLE
Move search box to lower left

### PR DESCRIPTION
**Pull request by lukemelia**
https://github.com/cardstack/boxel/assets/353/fdf3ddaa-8c47-4a80-a393-e2b24e6521ac

### CODE REVIEW

**Comment by backspace** [comment]
Percy shows this, backgrounds missing:

Build #640 - Cardstack - Percy | Visual testing as a service 2023-08-17 18-51-20

Preview deployment seems broken but running locally it does lack the split backgrounds:

Boxel 2023-08-17 19-00-06

Will look at the changes now to see if anything sticks out, or could it be unrelated to this PR somehow, a merge problem?

File: packages/host/app/components/operator-mode/container.gts [hunk 1]
```diff
@@ -532,7 +536,7 @@ export default class OperatorModeContainer extends Component<Signature> {
     ) {
       return htmlSafe(`background-image: url(${this.backgroundImageURLs[0]});`);
     }
-    return '';
```
Comment (backspace) [inline packages/host/app/components/operator-mode/container.gts:589]: I can’t explain it but removing this brings back the backgrounds:

Boxel 2023-08-17 19-16-39

File: packages/boxel-ui/addon/components/input/search-input/index.gts [hunk 2]
```diff
@@ -36,6 +36,7 @@ const SearchInput: TemplateOnlyComponent<Signature> = <template>
   >
     <label>
       <span class='boxel-sr-only'>Search</span>
+      {{! template-lint-disable no-inline-styles }}
```
Comment (backspace) [inline packages/boxel-ui/addon/components/input/search-input/index.gts]: Is `cssVar` a way around this?

File: packages/boxel-ui/addon/components/input/search-input/usage.gts [hunk 3]
```diff
@@ -13,8 +13,7 @@ export default class SearchInputUsage extends Component {
   @tracked bottomTreatment = this.defaultBottomTreatment;
   @tracked value = '';
   @tracked errorMessage = '';
-  @tracked placeholder =
-    'Boxel search input with validation.Boxel search input with validation.';
+  @tracked placeholder = 'Boxel search input with validation.';
```
Comment (backspace) [inline packages/boxel-ui/addon/components/input/search-input/usage.gts:16]: hah how did this happen 🧐

**Comment by backspace** [comment]
I don’t know if this is solvable in a general way but the alignment of these when in split view seems off to me:

Boxel 2023-08-17 19-22-40

File: packages/host/app/components/search-sheet/index.gts [hunk 4]
```diff
@@ -95,22 +96,15 @@ export default class SearchSheet extends Component<Signature> {
...
-    } else if (
       mode == SearchSheetMode.SearchPrompt ||
-      mode == SearchSheetMode.SearchResults
+      mode == SearchSheetMode.ChoosePrompt
     ) {
-      this._headline = 'Search cards and workspaces';
+      return 'Search for cards';
     }
-    return this._headline;
+    return 'Search for…';
```
Comment (backspace) [inline packages/host/app/components/search-sheet/index.gts:107]: true ellipsis: 🤩

File: packages/host/app/components/search-sheet/index.gts [hunk 5]
```diff
@@ -240,40 +235,58 @@ export default class SearchSheet extends Component<Signature> {
...
+      );
+    }
+    return htmlSafe('');
+  }
+
+  @action onSearchInputKeyDown(e: KeyboardEvent) {
+    if (e.key === 'Escape') {
+      this.onCancel();
+      (e.target as HTMLInputElement)?.blur?.();
+    }
```
Comment (backspace) [inline packages/host/app/components/search-sheet/index.gts:255]: This is annoying to exercise in tests but maybe worth it?

File: packages/host/app/components/search-sheet/index.gts [hunk 6]
```diff
@@ -329,120 +345,141 @@ export default class SearchSheet extends Component<Signature> {
...
+          height var(--boxel-transition),
+          width var(--boxel-transition),
+          font-size var(--boxel-transition),
+          border-radius var(--boxel-transition);
       }
 
       .closed {
-        height: var(--search-sheet-closed-height);
+        left: 20px;
+        height: calc(var(--search-sheet-closed-height) + 20px);
```
Comment (backspace) [inline packages/host/app/components/search-sheet/index.gts]: Are 40px/20px/10px and 172px extractable? Not worth it?

**Review by backspace — CHANGES_REQUESTED** [review]
The background thing is mysterious, otherwise looks good to me, some minor notes to consider

**Comment by habdelra** [comment]
So the background thing, I think  is a way that templates work. When an attribute (in this case “style”) is set in thr template, if its value is undefined then glimmer will omit the attribute. I believe that is why we are returning an empty string instead of undefined. Or at least that is my hunch without diving in to it more deeply

File: packages/host/app/components/search-sheet/index.gts (hunk 6 above)
Comment (burieberry) [reply packages/host/app/components/search-sheet/index.gts]: we can use --boxel-sp variables for the first three

**Review by burieberry — CHANGES_REQUESTED** [review]
Placing the search icon to the right conflicts with error validation icons, so we should take that into account if the SearchInput component will have validation capability

**Comment by backspace** [comment]
I suspect there won’t be a need for validation in that field so I’m leaving that for now. Or the icon could replace the magnifying glass when one exists?

This is something that predates the appearance change here but since this adds the ability to close the search with ⎋ I also think it should be possible to close it without that, maybe just by clicking outside the container? But this can be addressed separately.

File: packages/boxel-ui/addon/components/input/search-input/index.gts [hunk 7]
```diff
@@ -44,6 +45,10 @@ const SearchInput: TemplateOnlyComponent<Signature> = <template>
         @state={{if @state @state 'initial'}}
         @errorMessage={{@errorMessage}}
         @placeholder={{if @placeholder @placeholder 'Search'}}
+        style={{cssVar
+          input-height='var(--search-input-height)'
```
Comment (backspace) [inline packages/boxel-ui/addon/components/input/search-input/index.gts]: I changed this to `1000px` and it didn’t look different but I’m reluctant to remove it without understanding what it’s meant to do, any ideas @lukemelia?

Comment (lukemelia) [reply packages/boxel-ui/addon/components/input/search-input/index.gts]: Not needed. It's handled in the stylesheet. I'll remove

**Comment by lukemelia** [comment]
> This is something that predates the appearance change here but since this adds the ability to close the search with ⎋ I also think it should be possible to close it without that, maybe just by clicking outside the container?

I like this idea as a a future improvement

**Review by backspace — APPROVED** [review]

**Review by burieberry — APPROVED** [review]

**Pull request merged**